### PR TITLE
A CLI flag to allow fast transaction signing when account is unlocked.

### PR DESCRIPTION
**Pull request by tomusdrw**
Sending 250 txs: ~20s
Sending 250 txs with --fast-unlock: ~1.4s

### CODE REVIEW

**Comment by 5chdn** [comment]
Does this address #5739 ?

**Comment by tomusdrw** [comment]
Yes. But it doesn't really explain the difference in behaviour between 1.6.6 and 1.6.7. I'm in favour of closing this and re-opening if the problem is not fixed in 1.7.

File: ethcore/src/account_provider/mod.rs [hunk 1]
```diff
@@ -116,15 +116,22 @@ type AccountToken = String;
...
+	/// Dapps settings.
 	dapps_settings: RwLock<DappsSettingsStore>,
 	/// Accounts on disk
 	sstore: Box<SecretStore>,
 	/// Accounts unlocked with rolling tokens
 	transient_sstore: EthMultiStore,
 	/// Accounts in hardware wallets.
 	hardware_store: Option<HardwareWalletManager>,
+	/// When unlocking permanently on for some time store a raw secret instead of password.
+	fast_unlock: bool,
```
Comment (debris) [inline ethcore/src/account_provider/mod.rs]: it needs better description. I can't understand this one

**Comment by NikVolf** [comment]
Why providing another flag and not just storing private key instead of password, always, as we originally did for this unsafe unlocks?
Isn't storing private key provides less attack surface comparing to storing the password (that can be used to access multiple private keys)?

**Comment by tomusdrw** [comment]
> Isn't storing private key provides less attack surface comparing to storing the password (which can be used to access multiple private keys)?

Only if you assume that the attacker has read access to the filesystem (and knows where the private keys are). I think storing a password is strictly more secure, since it's not enough to scan memory/swapfile to get full access to your account.

In a very paranoid scenario you can even copy or mount a volume with private keys (encrypted or over the network) only when some transactions are supposed to be signed, while still having the account unlocked permantently.

@debris description updated.

**Pull request merged**